### ISSUE TITLE
Mod seems to simply not work: Fails to prevent item placement.  Lock icon does not show up.  Localization also broken.

### ISSUE THREAD

**Issue by SargeRyong** [hilburn/DimensionGuard#18]
Just installed DimensionGuard on a fresh 1.7.10 singleplayer client.

Configured my black list for:

>
    S:blacklist <
        minecraft:wool:*,0
     >

Start a new world, give myself a piece of orange wool in creative.  No lock icon.  Can place the block with no problem.

Swapped to survival, tried to place the block, was still able to.

Built a nether portal, went through, received message with broken localization:
 
> db.enabled.description: Orange Wool

Went back to overworld, received the "disabled" message, with broken localization.

> db.disabled.description: Orange Wool

Still no lock icon, still able to place the block.  

Using the following mod versions, on a brand new clean install, with no other mods:

Forge 10.13.4.1492
CodeChickenCore 1.0.6.43
DimensionGuard 1.2.2

**Comment by hilburn** [comment]
Please can you upload the console log when you start up the game through initialisation - don't need it beyond the start menu

**Comment by SargeRyong** [comment]
Sure thing.

Here's the log from launch to start menu: 

> [17:44:24] [main/INFO]: Setting user: YoraeRyong
[17:44:26] [Client thread/INFO]: LWJGL Version: 2.9.1
[17:44:37] [Client thread/INFO]: Reloading ResourceManager: Default, FMLFileResourcePack:Forge Mod Loader, FMLFileResourcePack:Minecraft Forge, FMLFileResourcePack:Dimension Guard
[17:44:38] [Sound Library Loader/INFO]: Sound engine started
[17:44:38] [Client thread/INFO]: Created: 16x16 textures/blocks-atlas
[17:44:38] [Client thread/INFO]: Created: 16x16 textures/items-atlas
[17:44:38] [Client thread/INFO]: Reloading ResourceManager: Default, FMLFileResourcePack:Forge Mod Loader, FMLFileResourcePack:Minecraft Forge, FMLFileResourcePack:Dimension Guard
[17:44:38] [Client thread/INFO]: Created: 512x256 textures/blocks-atlas
[17:44:39] [Client thread/INFO]: Created: 256x256 textures/items-atlas
[17:44:39] [Sound Library Loader/INFO]: Sound engine started

I don't see anything strange going on.  Did the same process as above just to see if there was anything interesting in the log in-world, but there wasn't:

> [17:47:08] [Server thread/INFO]: Starting integrated minecraft server version 1.7.10
[17:47:08] [Server thread/INFO]: Generating keypair
[17:47:08] [Server thread/INFO]: Preparing start region for level 0
[17:47:09] [Server thread/INFO]: Preparing spawn area: 74%
[17:47:09] [Server thread/INFO]: Changing view distance to 12, from 10
[17:47:10] [Server thread/INFO]: YoraeRyong[local:E:1e958879] logged in with entity id 331 at (49.64994312807918, 67.0, 271.79931930021013)
[17:47:10] [Server thread/INFO]: YoraeRyong joined the game
[17:47:41] [Server thread/INFO]: Saving and pausing game...
[17:47:41] [Server thread/INFO]: Saving chunks for level 'New World'/Overworld
[17:47:41] [Server thread/INFO]: Saving chunks for level 'New World'/Nether
[17:47:41] [Server thread/INFO]: Saving chunks for level 'New World'/The End
[17:47:52] [Client thread/INFO]: [CHAT] dg.enabled.description: Orange Wool
[17:48:12] [Client thread/INFO]: [CHAT] dg.disabled.description: Orange Wool
[17:48:17] [Server thread/INFO]: Saving and pausing game...

**Comment by hilburn** [comment]
There has to be more... forge logs even with no mods installed tend to run to a couple of hundred lines, I need the whole thing (on Pastebin or equivalent if possible)

**Comment by SargeRyong** [comment]
I pasted the forge "latest.log', which looks to be limited log messages of INFO or higher.  

fml-client-latest.log looks to contain everything, including debug and trace logs - however, it looks like it contains ONLY client-side logging and no server-side output.

Here's the client log from startup to menu screen: http://pastebin.com/LQFHZq8r

**Comment by hilburn** [comment]
Ok, as I thought, it's not working because the asm bytecode handlers aren't injecting properly - which is strange. I'll have to look into this more closely

**Comment by hilburn** [comment]
Also, with the broken lang files are you sure it was

>db.disabled.description: Orange Wool

not 

>d**g**.disabled.description: Orange Wool

because if it's the former then I have no explanation of how a string constant got corrupted like that, and Highly recommend you try re-downloading the file from Curse(Forge) just in case it got corrupted or something

**Comment by SargeRyong** [comment]
Oh, sorry about that, I must've made a typo when I pasted it.  It says "dg."0

**Comment by hilburn** [comment]
ok good, because that had me really worried ^^

I will try to find some time and/or energy this weekend to get this reproducing and then fixed for you

**Comment by PureSpider** [comment]
Any news on this? Would love to use this mod to prevent BC quarry usage destroying my overworld :-1:

**Comment by dillasyx** [comment]
no version is working...
lock not shown and not preventing from being placed and text has wrong format

**Comment by PureSpider** [comment]
Try [Item Blacklist](https://mods.curse.com/mc-mods/minecraft/237226-item-blacklist) instead, works flawlessly for me!
Make sure to read [the config instructions](https://github.com/DoubleDoorDevelopment/ItemBlacklist#pack-vs-world-bans) to have it working correctly (per world/globally).

**Comment by dillasyx** [comment]
S:blacklist <
        minecraft:stone,-28
        ImmersiveEngineering:woodenDevice:1,-28
        ImmersiveEngineering:woodenDevice:2,-28
        ImmersiveEngineering:woodenDevice:3,-28
     >

is not working

**Comment by PureSpider** [comment]
Did you read the config section I linked? You can not just move over the config of one mod and expect it to work with a different one...

**Comment by dillasyx** [comment]
you sent a config instruction of another mod

**Comment by PureSpider** [comment]
Yes, I also recommended a different mod that does the same as this mod and actually works.

**Comment by dillasyx** [comment]
ah thanks i see, i will try

**Comment by dillasyx** [comment]
are ranges only possible with dimensions, that are available? then it would take literally hours
can't go into world with:

{
  "__GLOBAL__": [],
  "-1021#44": [
    {
      "item": "ImmersiveEngineering:woodenDevice",
      "damage": "2"
    }
  ],
  "-1027#-1023": [
    {
      "item": "ImmersiveEngineering:woodenDevice",
      "damage": "2"
    }
  ],
  "-1029#-2546": [
    {
      "item": "ImmersiveEngineering:woodenDevice",
      "damage": "2"
    }
  ]
}

**Comment by PureSpider** [comment]
Would probably be a good idea to ask that in a ticket over at the other mod's github page, instead of here.
